tests/distributed_grids/2d_refinement_10.cc

```diff
@@ -0,0 +1,287 @@
+
+    for (unsigned int i = 0; n_cells_are_same && (i<n_cell_smooth.size()); ++i)
+      {
+        n_cells_are_same = n_cells_are_same &&
+                           (n_cell_smooth[i] == n_cell_no_smooth[i]);
+      }
+    deallog << "n_cells_are_same = " << n_cells_are_same << std::endl;
+  }
+  {
+    bool cell_center_loactions_are_same =
+      (final_cell_center_loactions_smooth.size() == final_cell_center_loactions_no_smooth.size());
+
+    std::set<Location<dim> >::const_iterator it1=final_cell_center_loactions_smooth.begin();
+    std::set<Location<dim> >::const_iterator it2=final_cell_center_loactions_no_smooth.begin();
+
+    for (; cell_center_loactions_are_same && (it1!=final_cell_center_loactions_smooth.end()); ++it1, ++it2)
+      {
+        cell_center_loactions_are_same = cell_center_loactions_are_same &&
+                                         // This is operator ==
+                                         (!(*it1 < *it2) && !(*it2 < *it1));
+      }
+    deallog << "cell_center_loactions_are_same = " << cell_center_loactions_are_same << std::endl;
+  }
+
+  deallog.pop();
+
+  return (0);
+}
+
+template<int dim>
+TriaTest<dim>::TriaTest(const typename dealii::Triangulation<dim>::MeshSmoothing smoothing_option)
+  :
+  mpi_communicator(MPI_COMM_WORLD),
+  triangulation(mpi_communicator, smoothing_option),
+  myid (Utilities::MPI::this_mpi_process (mpi_communicator)),
+  I_am_host(myid == 0)
+{
+  std::vector<unsigned int> repetitions;
+  Point<dim> p1;
+  Point<dim> p2;
+
+  for (unsigned int d=0; d<dim; ++d)
+    {
+      repetitions.push_back (2);
+      p1[d] = 0.0;
+      p2[d] = 1.0;
+    }
+  GridGenerator::subdivided_hyper_rectangle (triangulation,
+                                             repetitions,
+                                             p1,
+                                             p2);
+
+  // GridGenerator::hyper_cube(triangulation);
+  // triangulation.refine_global(1);
+}
+
+template<int dim>
+void TriaTest<dim>::run(std::vector<unsigned int> &n_cell,
+                        std::set<Location<dim> > &position_list)
+{
+  n_cell.clear();
+  position_list.clear();
+
+  unsigned int counter = 0;
+  if (I_am_host)
+    {
+      deallog << "n_loop  n_cell" << std::endl;
+      deallog << counter << "       "
+              << triangulation.n_global_active_cells() << std::endl;
+    }
+  ++counter;
+
+  for (; counter < 5; ++counter)
+    {
+      {
+        Point<dim> p;
+        for (unsigned int d=0; d<dim; ++d)
+          {
+            p[d] = 0.5 - std::pow(0.5, 1.0 + counter);
+          }
+        typename TypeTria::active_cell_iterator
+        cell = triangulation.begin_active();
+        for (; cell!= triangulation.end(); ++cell)
+          if (cell->is_locally_owned() && ((cell->center()).distance(p) < 1e-4))
+            {
+              cell->set_refine_flag();
+            }
+      }
+
+      triangulation.prepare_coarsening_and_refinement();
+      write_vtu(counter);
```

Comment:
Are you collecting these VTU files in the test output? If so, please use a format different than VTU (which is compressed and makes it difficult to figure out what the differences actually are). If you don't use these output files, just comment out these lines to avoid creating unnecessary output.